src/StreamSelectLoop.php

```diff
@@ -196,6 +237,9 @@ private function waitForStreamActivity($timeout)
         $write = $this->writeStreams;
 
         $available = $this->streamSelect($read, $write, $timeout);
+        if ($this->pcntl) {
+            \pcntl_signal_dispatch();
+        }
```

Comment:
Does this need to be checked after every loop iteration and does this cause any performance issues? It's my understanding that a signal would interrupt the `select()` call checked below, no?